detekt-rules-coroutines/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/coroutines/SuspendFunSwallowedCancellationSpec.kt

```diff
@@ -1272,9 +1273,9 @@ class SuspendFunSwallowedCancellationSpec(private val env: KotlinCoreEnvironment
         listOfStartLocation: List<SourceLocation>,
         listOfEndLocation: List<SourceLocation>,
     ) {
-        check(listOfEndLocation.size == listOfStartLocation.size)
-        assertThat(findings).hasSize(listOfStartLocation.size)
-        assertThat(findings).hasStartSourceLocations(*listOfStartLocation.toTypedArray())
-        assertThat(findings).hasEndSourceLocations(*listOfEndLocation.toTypedArray())
+        assertThat(listOfEndLocation).hasSameSizeAs(listOfStartLocation)
+        assertThatFindings(findings).hasSize(listOfStartLocation.size)
```

Comment:
```suggestion
        assertThatFindings(findings).hasSameSizeAs(listOfStartLocation)
```

Haven't tested, but assume this works with the custom assertion?